src/Scalar.jl

```diff
@@ -0,0 +1,20 @@
+"""
+    Scalar{T}(x::T)
+
+Construct a statically-sized 0-dimensional array that contains a single element,
+`x`. This type is particularly useful for influencing broadcasting operations.
+"""
+immutable Scalar{T} <: StaticArray{T,0}
+    data::T
+end
+
+@inline (::Type{Scalar}){T}(x::Tuple{T}) = Scalar{T}(x[1])
```

Comment:
I got caught up on this too, but I think I'm ok after reading Andy's comment - `Tuple{T}` being the tuple containing exactly one `T` (as quite distinct from `Array{T}` containing many `T`).